File: Projects_Javascript/JS/project.js
function Project() {
    this.name = "";
    this.allocation = 0;
    this.startdate = "";
    this.enddate = "";
    this.status = "Open";



}

Project.prototype.addProject = function() {
    this.name = $('projectname').element.value;
    this.startdate = $('startdate').element.value;
    this.enddate = $('enddate').element.value;
    this.allocation = $('allocation').element.value;
}

Project.prototype.changeStatus = function(status) {
    this.status = status;
}

Project.prototype.setProject = function(obj) {
    this.name = obj.name;
    this.allocation = obj.allocation;
    this.startdate = obj.startDate;
    this.enddate = obj.endDate;
    this.status = obj.status;
}

File: Angular - 2/ProjectsAPI/ProjectApi.Contracts/Project.cs
using System;

namespace ProjectsAPI.Models
{
    public class Project
    {
        public Project()
        {

        }
        public Project(string name, int allocation, DateTime startDate, DateTime endDate, string status)
        {
            this.Name = name;
            this.Allocation = allocation;
            this.StartDate = startDate;
            this.EndDate = endDate;
            this.Status = status;
        }

        public string Name { get; set; }
        public int Allocation { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string Status { get; set; }

        internal void Update(Project project)
        {
            this.Name = project.Name;
            this.Allocation = project.Allocation;
            this.StartDate = project.StartDate;
            this.EndDate = project.EndDate;
            this.Status = project.Status;
        }
    }
}